File: dynamic_programming/pascal_triangle2/solution.py
def getRow(rowIndex):
    row = [1]
    for i in range(1, rowIndex + 1):
        row.append(1)
        for j in range(i - 1, 0, -1):
            row[j] = row[j] + row[j - 1]
    return row

# Example Usage
print(getRow(4))  # Output: [1, 4, 6, 4, 1]

File: dynamic_programming/pascal_triangle2/solution.js
var getRow = function (rowIndex) {
  let row = [1];
  for (let i = 1; i <= rowIndex; i++) {
    row.push(1); // Add a new 1 at the end
    for (let j = i - 1; j > 0; j--) {
      row[j] = row[j] + row[j - 1];
    }
  }
  return row;
};

// Example Usage
console.log(getRow(4)); // Output: [1, 4, 6, 4, 1]

console.log(getRow(5)); // Output: [1, 5, 10, 10, 5, 1]

